app/models/list_import/anilist/row.rb

```diff
@@ -1,98 +1,126 @@
+
+      attr_reader :node, :type
 
-        Mapping.lookup('anilist', key) ||
-          Mapping.guess(type.classify.safe_constantize, media_info)
+      def fields
+        %i[
+          rating status reconsume_count progress
+          notes started_at finished_at
+        ]
       end
 
+      # For mapping guess
       def media_info
         {
-          id: node[type]['id'],
-          title: node[type]['title_romaji'],
-          subtype: node[type]['type'],
-          episode_count: node[type]['total_episodes'],
-          chapter_count: node[type]['total_chapters']
+          title: title,
+          subtype: type,
+          episode_count: media.episodes,
+          chapter_count: media.chapters
         }.compact
       end
 
+      # 100-point scale to 20-point scale (raw)
+      # rating -> score
+      def rating
+        return nil if node.score.zero?
+
+        [(node.score.to_f / 5).ceil, 2].max
+      end
+
       def status
-        case node['list_status']
+        case node.status.downcase
         when 'completed' then :completed
-        when /ing\z/ then :current
-        when /\Aplan/ then :planned
-        when 'on-hold' then :on_hold
+        when 'current' then :current
+        when 'planning' then :planned
+        when 'paused' then :on_hold
         when 'dropped' then :dropped
         end
       end
 
-      def progress
-        type == 'anime' ? node['episodes_watched'] : node['chapters_read']
+      # reconsume_count -> repeat
+      def reconsume_count
+        node.repeat
       end
 
-      def volumes
-        return unless type == 'manga'
-
-        node['volumes_read']
+      def progress
+        node.progress
       end
 
-      def rating
-        return nil if node['score_raw'].zero?
-
-        # 100-point scale to 20-point scale (raw)
-        [(node['score_raw'].to_f / 5).ceil, 2].max
+      def notes
+        node.notes
       end
 
-      def reconsume_count
-        type == 'anime' ? node['rewatched'] : node['reread']
-      end
+      # 2020-05-19 -> yyyy-mm-dd
+      def started_at
+        return if node.started_at.to_h.compact.blank?
 
-      def notes
-        node['notes']
+        # will properly convert to include 0 before days/months
+        formatted_date(node.started_at)
       end
 
-      def started_at
-        return nil if node['started_on'].nil?
+      # finished_at -> completed_at
+      def finished_at
+        return if node.completed_at.to_h.compact.blank?
 
-        date = node['started_on'].split('/')
+        # will properly convert to include 0 before days/months
+        formatted_date(node.completed_at)
+      end
 
-        DateTime.strptime(format_date(date), '%F')
-      rescue
-        nil
+      def formatted_date(date_node)
+        date_node.to_h.symbolize_keys.values_at(:year, :month, :day).join('-').to_date.to_s
       end
 
-      def finished_at
-        return nil if node['finished_on'].nil?
+      def anilist_key
+        "AniList/#{type}"
```

Comment:
Can we do lower case `anilist/#{type}` to be consistent?

Reply:
sure